lib/internal/fs/glob.js

```diff
@@ -36,7 +38,22 @@ class Cache {
   #cache = new SafeMap();
   #statsCache = new SafeMap();
   #readdirCache = new SafeMap();
+  #opendirCache = new SafeMap();
 
+  async stat(path) {
+    const cached = this.#statsCache.get(path);
+    if (cached) {
+      return cached;
+    }
+    let val;
+    try {
+      val = await lstat(path);
+    } catch {
+      val = null;
+    }
+    this.#statsCache.set(path, val);
+    return val;
```

Comment:
```suggestion
	const promise = PromisePrototypeThen(lstat(path), null, () => null);
    this.#statsCache.set(path, promise);
    return promise;
```
to avoid race condition, cache promise and not its result